homeassistant/components/hydrawise/entity.py

```diff
@@ -24,24 +24,38 @@ def __init__(
         coordinator: HydrawiseDataUpdateCoordinator,
         description: EntityDescription,
         controller: Controller,
+        *,
         zone: Zone | None = None,
+        sensor: Sensor | None = None,
     ) -> None:
         """Initialize the Hydrawise entity."""
         super().__init__(coordinator=coordinator)
         self.entity_description = description
         self.controller = controller
-        self.zone = zone
-        self._device_id = str(controller.id if zone is None else zone.id)
+        self.zone_id = zone.id if zone else None
+        self.sensor_id = sensor.id if sensor else None
+        self._device_id = str(zone.id) if zone is not None else str(controller.id)
         self._attr_unique_id = f"{self._device_id}_{description.key}"
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, self._device_id)},
-            name=controller.name if zone is None else zone.name,
+            name=zone.name if zone is not None else controller.name,
+            model="Zone" if zone is not None else controller.hardware.model.description,
             manufacturer=MANUFACTURER,
         )
-        if zone is not None:
+        if zone is not None or sensor is not None:
             self._attr_device_info["via_device"] = (DOMAIN, str(controller.id))
         self._update_attrs()
 
+    @property
+    def zone(self) -> Zone | None:
+        """Return the entity zone."""
+        return self.coordinator.data.zones[self.zone_id] if self.zone_id else None
```

Comment:
```suggestion
    def zone(self) -> Zone:
        """Return the entity zone."""
        assert self.zone_id 
        return self.coordinator.data.zones[self.zone_id]
```

Reply:
Won't work like that. There are many places where we check whether the sensor is part of a zone or not by comparing it to `None`.

Reply:
Can we replace that by checking with `self.zone_id`?

This way you can get rid of a lot of asserts